### ISSUE TITLE
java.lang.StackOverflowError building Macaca_fascicularis

### ISSUE THREAD

**Issue by mjafin** [pcingola/SnpEff#133]
Hi Pablo,
I noticed there was no db for snpeff 4.2, Macaca_fascicularis. I followed your way of preparing things (https://github.com/pcingola/SnpEff/tree/master/scripts_build/data/Macaca_fascicularis) and get this:
```
                880000  ....................................................................................................
                890000  ....................................................................................................
                900000  .........................................java.lang.StackOverflowError
        at java.lang.String.toLowerCase(String.java:2519)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.hasAttr(GffMarker.java:207)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:117)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGffParentId(GffMarker.java:140)
        at ca.mcgill.mcb.pcingola.interval.GffMarker.getGeneId(GffMarker.java:119)
```
I tried increasing stack size but memory consumption goes as high as I allow it to and then crashes. Infinite loop maybe?

I'll look at the gff file at around 904.1k (judging by the above 900000 and 41 dots, very scientific) and will let you know if there's a funny gene in there

**Comment by fabfabBesnard** [comment]
I had the same issue with an Arabidopsis genome build ([consolidated_annotation.Ws_0.gff3.bz2](http://mus.well.ox.ac.uk/19genomes/annotations/consolidated_annotation_9.4.2011/gene_models/)).
Except that in my cases, it crashes from start, after only 19 dots !

FabFab

**Comment by pcingola** [comment]
Fixed on dev. version (4.3).

**Comment by fabfabBesnard** [comment]
Great !

Thanks a lot,

best,
Fabrice

Le 04/07/2016 23:26, Pablo Cingolani a écrit :
>
> Closed #133 <https://github.com/pcingola/SnpEff/issues/133>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/pcingola/SnpEff/issues/133#event-712391619>, or 
> mute the thread 
> <https://github.com/notifications/unsubscribe/ATPXwq288LCkV6rG4VTcvZQbiEb7iYJLks5qSXqCgaJpZM4H7eCr>.
>

-- 
Fabrice Besnard
Laboratoire Reproduction et Développement des Plantes (RDP)
Ecole Normale Supérieure de Lyon (ENS-Lyon)
46, allée d'Italie, 69364 LYON Cedex 07.
Building M7, Ground floor. Office: Room 10.
mail: fabrice.besnard@ens-lyon.fr
Tel: +33-4-72-72-86-05
